### PR TITLE
Blazor catch-all parameters

### LINKED ISSUE #19632: [RC1] Add documentation on catch-all parameters
https://github.com/dotnet/aspnetcore/issues/25199#issuecomment-679321134

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 35474d24-8039-f04a-de14-1530b8e8506c
* Version Independent ID: b6cb2574-6326-a40d-0b6c-d1661328e828
* Content: [ASP.NET Core Blazor routing](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-3.1)
* Content Source: [aspnetcore/blazor/fundamentals/routing.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/fundamentals/routing.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #19632

Based on:

* https://github.com/dotnet/aspnetcore/issues/25199#issuecomment-679321134
* https://github.com/dotnet/aspnetcore/pull/24038#issuecomment-660154330

### CODE REVIEW

**Comment by zbecknell** [comment]
So initially when implementing this I was confused by looking at MVC routing and thought that `**` should be implemented as part of this feature.  Turns out the `**` was specifically talking about URL _building_ rather than actual routing.

In the end, the current Blazor router only allows `*` -- we will throw for more than one `*` (see `TemplateSegment.cs:25`) to protect for potentially using `**` in the future.

So in other words, we don't need to mention `**` or anything to do with encoding/decoding, as whatever is captured in the catch-all parameter should come through as-is.

**Comment by guardrex** [comment]
@zbecknell ... Sorry that I didn't read the whole convo over there. It was a **_TL;DR_** to me at like 4am! :smile: lol

> should come through as-is

... but I thought you said that **_you do decode at_** ...

https://github.com/dotnet/aspnetcore/pull/24038#discussion_r456513976

**Comment by zbecknell** [comment]
No worries!  And now I'm adding to the confusion!  I shouldn't have said "as-is" -- a catch-all won't be decoded any differently than other params.  So  for `/blog/{*parameter}`, `/blog/value1/foo%20bar` -> `parameter == "value1/foo bar"`

**Comment by guardrex** [comment]
Is a slash included? For `/blog/value1/foo%20bar`, which would be captured into the var ...

* `value1/foo bar`
* `/value1/foo bar`

**Comment by zbecknell** [comment]
Negative, the first slash is part of the template, so it's not included.

**Comment by guardrex** [comment]
Yes ... I see what you mean. `/blog/{*parameter}` ... "/blog/" ... gotcha. Thanks.

**Comment by guardrex** [comment]
kk ... I think we're good here. We'll just wait and see when Artak wants these RC1 PRs merged ... now or at RC1 release.

**Comment by guardrex** [comment]
We're going to temporarily surface it this way (see the diff). I'll update this when RC1 releases and then again at .NET 5 GA.
